### DISCUSSION TITLE
Docusaurus 添加评论功能 | Alan Wang

### DISCUSSION THREAD

**Discussion by giscus** [3Alan/site#7] [Announcements]
# Docusaurus 添加评论功能 | Alan Wang

通过 Docusaurus Swizzling 为博客和文档添加 Giscus 评论功能

https://www.alanwang.site/posts/%E5%8D%9A%E5%AE%A2%E6%90%AD%E5%BB%BA%E7%B3%BB%E5%88%97/docusaurus-%E8%AF%84%E8%AE%BA%E7%B3%BB%E7%BB%9F

**Comment by b9348** [comment]
所有页面公用一份评论区？

**Reply by 3Alan** [reply]
每篇文章都有自己的评论，不是全局的

**Reply by b9348** [reply]
> 每篇文章都有自己的评论，不是全局的

https://b9348.gitee.io/ 用手机第一次打开以后评论区加载不出来最底下的，必须翻页一次才能加载出来。咋办啊大佬请问你知道咋整嘛0.0 用电脑的都是正常。按F12反复看有时候电脑也会，那个时候评论的网络请求是404了

**Reply by 3Alan** [reply]
我自己网站看着没问题，你可能哪里配置不对，你仓库地址发下我看看你代码

**Reply by b9348** [reply]
> 我自己网站看着没问题，你可能哪里配置不对，你仓库地址发下我看看你代码

啊这 刚刚才看到 。。https://gitee.com/b9348/b9348.git ，是仓库。电脑稳定复现的方法我刚刚发现了，用新标签页窗口化，窗口宽度用鼠标拖一下，伸缩到最小以后再打开https://b9348.gitee.io/ ，然后评论区无论如何都加载不出来，除非点下一页。本地跑代码是没有任何问题的。部署上去以后就有这个问题。分支是reactpress

**Reply by 3Alan** [reply]
移动端的时候 Giscus 的父元素隐藏了，解决办法：我猜你可能想让评论的宽度为100%？那就改成这样好了，或者你直接对着我文章里的复制也行（评论的宽度和文章的宽度一致）
![image](https://user-images.githubusercontent.com/49978973/197668082-c82e24ef-a1a4-4fa8-9f48-3cf100ab9f58.png)
```jsx
<>
  <div className="row">
    <div className={clsx('col', !docTOC.hidden && styles.docItemCol)}>
      <DocVersionBanner />
      <div className={styles.docItemContainer}>
        <article>
          <DocBreadcrumbs />
          <DocVersionBadge />
          {docTOC.mobile}
          <DocItemContent>{children}</DocItemContent>
          <DocItemFooter />
        </article>
        <DocItemPaginator />
      </div>
    </div>
    {docTOC.desktop && <div className="col col--3">{docTOC.desktop}</div>}
  </div>
  <div className={styles.last}>
    <br />
    <div className={styles.box}>{!hideComment && <Comment />}</div>
  </div>
</>
```

**Reply by b9348** [reply]
搞好了谢谢大佬！！！

**Comment by Fionn96** [comment]
感谢大佬！您的网站好像只有 dark 模式，请问如何自定义 Giscus 跟随网站 dark 和 light 模式自动转变 theme 主题呢？

**Reply by 3Alan** [reply]
已添加切换 dark 模式功能，你可以查看最新文章

**Reply by Fionn96** [reply]
已搞定，并且根据您的[Docusaurus DocSearch 搭建](https://www.alanwang.site/posts/%E5%8D%9A%E5%AE%A2%E6%90%AD%E5%BB%BA%E7%B3%BB%E5%88%97/Docusaurus-%E6%90%9C%E7%B4%A2)文章，成功配置全文搜索功能。再次感谢😄

**Reply by 3Alan** [reply]
不客气，很高兴能帮到你

**Comment by tianrking** [comment]
HI

**Comment by Shake-Jarjane** [comment]
大佬我发现一个问题，比如我点来时当前文档的界面，然后当前文档的评论也加载出来了，当我在左边目录点击别的文档的时候，另一个文档的评论内容会一直是当前文档的评论内容，需要刷新网页才改变，这个有办法解决吗？

**Reply by 3Alan** [reply]
回家过年了，年后再看看😂

**Reply by Shake-Jarjane** [reply]
ok哈哈哈

**Reply by 3Alan** [reply]
问题修复了，文章已经更新✅

**Reply by Shake-Jarjane** [reply]
按着步骤来网页崩了😂

类型“Window & typeof globalThis”上不存在属性“emitter”。ts(2339)

Cannot read properties of undefined (reading 'on')

**Reply by 3Alan** [reply]
把你代码仓库给我，这点信息我也没法定位错误

**Reply by Shake-Jarjane** [reply]
https://github.com/Shake-Jarjane/Docusaurus-Blog，这个是上个版本的commit

**Reply by 3Alan** [reply]
我这边看着没什么问题，你如果是复制了我的代码应该是不会有问题的，你检查一下是不是哪里你自己改动过

**Reply by Shake-Jarjane** [reply]
可以了，非常感谢大佬的帮助！

**Comment by lookmori** [comment]
大佬，我的这个错误怎么解决？
[404](http://www.mixly.lookmori.cn/blog/)
我找到一个[issues](https://github.com/giscus/giscus/issues/1268#issue-2057108719),我试了一下重新关闭discussion还是不行

**Reply by lookmori** [reply]
好了，添加一条评论就可以😂，谢谢
